src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/RecognisePriorLearningRequestToDetailsViewModelMapper.cs

```diff
@@ -25,12 +25,54 @@ public async Task<PriorLearningDetailsViewModel> Map(RecognisePriorLearningReque
                 DraftApprenticeshipId = source.DraftApprenticeshipId,
                 ProviderId = source.ProviderId,
                 DraftApprenticeshipHashedId = source.DraftApprenticeshipHashedId,
+                WeightageReducedBy = apprenticeship.WeightageReducedBy,
+                QualificationsForRplReduction = apprenticeship.QualificationsForRplReduction,
+                ReasonForRplReduction = apprenticeship.ReasonForRplReduction,
+                DurationReducedByHours = apprenticeship.DurationReducedByHours,
                 ReducedDuration = apprenticeship.DurationReducedBy,
                 ReducedPrice = apprenticeship.PriceReducedBy,
+            };
+        }
+    }
+
+    public class RecognisePriorLearningRequestToDataViewModelMapper : IMapper<RecognisePriorLearningRequest, PriorLearningDataViewModel>
+    {
+        private readonly ICommitmentsApiClient _commitmentsApiClient;
+
+        public RecognisePriorLearningRequestToDataViewModelMapper(ICommitmentsApiClient commitmentsApiClient)
+        {
+            _commitmentsApiClient = commitmentsApiClient;
+        }
+
+        public async Task<PriorLearningDataViewModel> Map(RecognisePriorLearningRequest source)
+        {
+            var apprenticeship = await _commitmentsApiClient.GetDraftApprenticeship(source.CohortId, source.DraftApprenticeshipId);
```

Comment:
This is a direct call to the inner api, and should be replaced by a call to an outer api endpoint that can provide all of the information required to create the viewmodel.

Reply:
@chrisfoster76 this might be a repeat comment, but I appeared to have lost my last message I posted

We call the outer API to get the RPL Data, and we didn't pollute the intent/purpose of that API call with information needed for navigation purposes. Therefore we make a second call (which previously exists) to the inner API to get the  apprenticeship details. 

The question is do we write API endpoints with the requirements of the calling program in place or do we write them a clear intent/purpose.  

Can we chat about this in the morning